libffi fails to build on Windows on ARM using tools-windows-msvc

We are somewhere injecting an additional -m64 flag `-DFFI_BUILDING_DLL -m64` which is used by libffi's "msvcc.sh" to configure x86_64 environments. This causes the armasm64 assembler to choke on some macros.

Building it manually works fine:
```sh
msys2_shell -defterm -no-start -msys2 -full-path -here

./configure \
  --build=aarch64-pc-cygwin --host=aarch64-pc-cygwin \
  --prefix="/c/GNUstep/arm64/Debug" \
  --disable-docs \
  CC="$PWD/msvcc.sh -g -marm64 -DUSE_DEBUG_RTL" CXX="$PWD/msvcc.sh  -g -marm64 -DUSE_DEBUG_RTL" LD=link \
  CPP="cl -nologo -EP" CXXCPP="cl -nologo -EP" \
  CPPFLAGS="-DFFI_BUILDING_DLL" LDFLAGS=""
```

```sh
$ make -j 8
MAKE x86_64-pc-msys : 0 * all-all
make[1]: Entering directory '/c/tools-windows-msvc/src/libffi/aarch64-pc-cygwin'
make  all-recursive
make[2]: Entering directory '/c/tools-windows-msvc/src/libffi/aarch64-pc-cygwin'
Making all in include
make[3]: Entering directory '/c/tools-windows-msvc/src/libffi/aarch64-pc-cygwin/include'
make[3]: Nothing to be done for 'all'.
make[3]: Leaving directory '/c/tools-windows-msvc/src/libffi/aarch64-pc-cygwin/include'
Making all in testsuite
make[3]: Entering directory '/c/tools-windows-msvc/src/libffi/aarch64-pc-cygwin/testsuite'
make[3]: Nothing to be done for 'all'.
make[3]: Leaving directory '/c/tools-windows-msvc/src/libffi/aarch64-pc-cygwin/testsuite'
Making all in man
make[3]: Entering directory '/c/tools-windows-msvc/src/libffi/aarch64-pc-cygwin/man'
make[3]: Nothing to be done for 'all'.
make[3]: Leaving directory '/c/tools-windows-msvc/src/libffi/aarch64-pc-cygwin/man'
make[3]: Entering directory '/c/tools-windows-msvc/src/libffi/aarch64-pc-cygwin'
source='../src/aarch64/win64_armasm.S' object='src/aarch64/win64_armasm.lo' libtool=yes \
DEPDIR=.deps depmode=none /bin/sh ../depcomp \
/bin/sh ./libtool  --tag=CC   --mode=compile /c/tools-windows-msvc/src/libffi/msvcc.sh -g -marm64 -DHAVE_CONFIG_H -I. -I..  -I. -I../include -Iinclude -I../src -DFFI_BUILDING_DLL -I. -I../include -Iinclude -I../src  -c -o src/aarch64/win64_armasm.lo ../src/aarch64/win64_armasm.S
libtool: compile:  /c/tools-windows-msvc/src/libffi/msvcc.sh -g -marm64 -DHAVE_CONFIG_H -I. -I.. -I. -I../include -Iinclude -I../src -DFFI_BUILDING_DLL -I. -I../include -Iinclude -I../src -c ../src/aarch64/win64_armasm.S  -DDLL_EXPORT -DPIC -o src/aarch64/.libs/win64_armasm.obj
win64_armasm.S
libtool: compile:  /c/tools-windows-msvc/src/libffi/msvcc.sh -g -marm64 -DHAVE_CONFIG_H -I. -I.. -I. -I../include -Iinclude -I../src -DFFI_BUILDING_DLL -I. -I../include -Iinclude -I../src -c ../src/aarch64/win64_armasm.S -o src/aarch64/win64_armasm.obj >/dev/null 2>&1
/bin/sh ./libtool  --tag=CC   --mode=link /c/tools-windows-msvc/src/libffi/msvcc.sh -g -marm64  -O2   -o libffi_convenience.la  src/prep_cif.lo src/types.lo src/raw_api.lo src/java_raw_api.lo src/closures.lo src/tramp.lo   src/aarch64/ffi.lo src/aarch64/win64_armasm.lo
/bin/sh ./libtool  --tag=CC   --mode=link /c/tools-windows-msvc/src/libffi/msvcc.sh -g -marm64  -O2 -no-undefined -version-info `grep -v '^#' ../libtool-version`     -o libffi.la -rpath /c/GNUstep/arm64/Debug/lib src/prep_cif.lo src/types.lo src/raw_api.lo src/java_raw_api.lo src/closures.lo src/tramp.lo  src/aarch64/ffi.lo src/aarch64/win64_armasm.lo
libtool: link: lib -OUT:.libs/libffi_convenience.lib src/.libs/prep_cif.obj src/.libs/types.obj src/.libs/raw_api.obj src/.libs/java_raw_api.obj src/.libs/closures.obj src/.libs/tramp.obj src/aarch64/.libs/ffi.obj src/aarch64/.libs/win64_armasm.obj
Microsoft (R) Library Manager Version 14.39.33321.0
Copyright (C) Microsoft Corporation.  All rights reserved.

libtool: link: ( cd ".libs" && rm -f "libffi_convenience.la" && cp -pR "../libffi_convenience.la" "libffi_convenience.la" )
libtool: link: /c/tools-windows-msvc/src/libffi/msvcc.sh -g -marm64 -o .libs/libffi-8.dll  src/.libs/prep_cif.obj src/.libs/types.obj src/.libs/raw_api.obj src/.libs/java_raw_api.obj src/.libs/closures.obj src/.libs/tramp.obj src/aarch64/.libs/ffi.obj src/aarch64/.libs/win64_armasm.obj   -g -marm64 -O2   `func_echo_all "" | /usr/bin/sed 's/ -lc$//'` -link -dll
libtool: link: linknames=
libtool: link: true
libtool: link: ( cd ".libs" && rm -f "libffi.la" && cp -pR "../libffi.la" "libffi.la" )
make[3]: Leaving directory '/c/tools-windows-msvc/src/libffi/aarch64-pc-cygwin'

```

<details>
<summary>Failed Build</summary>
```sh
libtool: compile:  /c/tools-windows-msvc/src/libffi/msvcc.sh -g -marm64 -DUSE_DEBUG_RTL -DHAVE_CONFIG_H -I. -I.. -I. -I../include -Iinclude -I../src -DFFI_BUILDING_DLL -m64 -c ../src/closures.c -o src/closures.obj >/dev/null 2>&1
C:/tools-windows-msvc/src/libffi/src/aarch64/ffi.c(314): warning C4146: unary minus operator applied to unsigned type, result still unsigned
win64_armasm.S
libtool: compile:  /c/tools-windows-msvc/src/libffi/msvcc.sh -g -marm64 -DUSE_DEBUG_RTL -DHAVE_CONFIG_H -I. -I.. -I. -I../include -Iinclude -I../src -DFFI_BUILDING_DLL -m64 -c ../src/aarch64/ffi.c -o src/aarch64/ffi.obj >/dev/null 2>&1
 Assembling: src/aarch64/.libs/win64_armasm.asm
src/aarch64/.libs/win64_armasm.asm(1062) : error A2008:syntax error : OPT
src/aarch64/.libs/win64_armasm.asm(3089) : error A2008:syntax error : GBLA
src/aarch64/.libs/win64_armasm.asm(3092) : error A2008:syntax error : GBLA
src/aarch64/.libs/win64_armasm.asm(3093) : error A2008:syntax error : GBLA
src/aarch64/.libs/win64_armasm.asm(3094) : error A2008:syntax error : GBLA
src/aarch64/.libs/win64_armasm.asm(3095) : error A2008:syntax error : GBLA
src/aarch64/.libs/win64_armasm.asm(3096) : error A2008:syntax error : GBLS
src/aarch64/.libs/win64_armasm.asm(3097) : error A2008:syntax error : GBLS
src/aarch64/.libs/win64_armasm.asm(3100) : error A2008:syntax error : GBLS
src/aarch64/.libs/win64_armasm.asm(3101) : error A2008:syntax error : GBLL
src/aarch64/.libs/win64_armasm.asm(3104) : error A2008:syntax error : GBLS
src/aarch64/.libs/win64_armasm.asm(3105) : error A2008:syntax error : GBLS
src/aarch64/.libs/win64_armasm.asm(3106) : error A2008:syntax error : GBLA
src/aarch64/.libs/win64_armasm.asm(3107) : error A2008:syntax error : GBLS
src/aarch64/.libs/win64_armasm.asm(3108) : error A2008:syntax error : GBLS
src/aarch64/.libs/win64_armasm.asm(3109) : error A2008:syntax error : GBLS
src/aarch64/.libs/win64_armasm.asm(3110) : error A2008:syntax error : GBLS
src/aarch64/.libs/win64_armasm.asm(3111) : error A2008:syntax error : GBLL
src/aarch64/.libs/win64_armasm.asm(3112) : error A2008:syntax error : GBLA
src/aarch64/.libs/win64_armasm.asm(3113) : error A2008:syntax error : GBLS
src/aarch64/.libs/win64_armasm.asm(3114) : error A2008:syntax error : GBLS
src/aarch64/.libs/win64_armasm.asm(3123) : error A2008:syntax error : MACRO
src/aarch64/.libs/win64_armasm.asm(3124) : error A2008:syntax error : __EmitRunningLabelAndOpcode
src/aarch64/.libs/win64_armasm.asm(3126) : error A2008:syntax error : SETS
src/aarch64/.libs/win64_armasm.asm(3127) : error A2008:syntax error : SETA
src/aarch64/.libs/win64_armasm.asm(3129) : error A2044:invalid character in file
src/aarch64/.libs/win64_armasm.asm(3130) : error A2008:syntax error : $__RunningLabel
src/aarch64/.libs/win64_armasm.asm(3131) : error A2044:invalid character in file
src/aarch64/.libs/win64_armasm.asm(3132) : error A2008:syntax error : $__RunningLabel
src/aarch64/.libs/win64_armasm.asm(3133) : error A2044:invalid character in file
src/aarch64/.libs/win64_armasm.asm(3134) : error A2008:syntax error : $__RunningLabel
src/aarch64/.libs/win64_armasm.asm(3135) : error A2044:invalid character in file
src/aarch64/.libs/win64_armasm.asm(3136) : error A2008:syntax error : $__RunningLabel
src/aarch64/.libs/win64_armasm.asm(3137) : error A2044:invalid character in file
src/aarch64/.libs/win64_armasm.asm(3138) : error A2008:syntax error : $__RunningLabel
src/aarch64/.libs/win64_armasm.asm(3139) : error A2044:invalid character in file
src/aarch64/.libs/win64_armasm.asm(3140) : error A2008:syntax error : $__RunningLabel
src/aarch64/.libs/win64_armasm.asm(3141) : fatal error A1010:unmatched block nesting
make[3]: *** [Makefile:1299: src/aarch64/win64_armasm.lo] Error 1
make[3]: *** Waiting for unfinished jobs....
make[3]: Leaving directory '/c/tools-windows-msvc/src/libffi/aarch64-pc-cygwin'
make[2]: *** [Makefile:1397: all-recursive] Error 1
make[2]: Leaving directory '/c/tools-windows-msvc/src/libffi/aarch64-pc-cygwin'
make[1]: *** [Makefile:625: all] Error 2
make[1]: Leaving directory '/c/tools-windows-msvc/src/libffi/aarch64-pc-cygwin'
make: *** [Makefile:593: all-all] Error 2
Failed
</details>

--- a/phases/13-libffi.sh
+++ b/phases/13-libffi.sh
@@ -47,7 +47,7 @@ rm -rf $TARGET
   --disable-docs \
   CC="$MSVCC" CXX="$MSVCC" LD=link \
   CPP="cl -nologo -EP" CXXCPP="cl -nologo -EP" \
-  CPPFLAGS="-DFFI_BUILDING_DLL" LDFLAGS="" \
+  CPPFLAGS="-DFFI_BUILDING_DLL" LDFLAGS="" CFLAGS="" \
 
 echo
 echo "### Building"

--- a/phases/13-libffi.sh
+++ b/phases/13-libffi.sh
@@ -47,7 +47,7 @@ rm -rf $TARGET
   --disable-docs \
   CC="$MSVCC" CXX="$MSVCC" LD=link \
   CPP="cl -nologo -EP" CXXCPP="cl -nologo -EP" \
-  CPPFLAGS="-DFFI_BUILDING_DLL" LDFLAGS="" CFLAGS="" \
+  CPPFLAGS="-DFFI_BUILDING_DLL" CFLAGS="" LDFLAGS="" \
 
 echo
 echo "### Building"

